File: String/37. Minimum characters to be added at front to make it palindrome.cpp
    bool ispali(string tmp){
        int i = 0, j = tmp.size()-1;
        while(i <= j){
            if(tmp[i] != tmp[j])return false;
            else {
                i++;
                j--;
            }
        }
        return true;
    }
    int minChar(string str){
        //Write your code here
        int count = 0;
        for(int i = 0; i < str.size(); i++){
            string tmp = str.substr(0, i+1);
            if(ispali(tmp)){
                count = max(count, i+1);
            }
        }
        return str.size() - count;
    }
};

File: String/37. Minimum characters to be added at front to make it palindrome.java
class Solution {
    public static int minChar(String str) {
       //Write your code here
       int count = 0;
       int low = 0;
       int high = str.length()-1;
       while(low < high){
           if(str.charAt(low) == str.charAt(high)){
               low++;
               high--;
           }
           else{
               count++;
               low = 0;
               high = str.length()-1-count;
           }
       }
       return count;
    }
}
